core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2834,6 +2834,28 @@ public static ByteString bitAnd(ByteString b0, ByteString b1) {
     return binaryOperator(b0, b1, (x, y) -> (byte) (x & y));
   }
 
+  /** Helper function for implementing <code>BITCOUNT</code>. Counts the number
+   * of bits set in an integer value. */
+  public static long bitCount(long b) {
+    return Long.bitCount(b);
+  }
+
+  /** Helper function for implementing <code>BITCOUNT</code>. Counts the number
+   * of bits set in the integer portion of a decimal value. */
+  public static long bitCount(BigDecimal b) {
+    return Long.bitCount(b.longValue());
```

Comment:
This also rounds down the decimal number, which is not obvious.
Please also note that a decimal may not fit into a long since we merged https://github.com/apache/calcite/pull/3589
So I am not sure this implementation is correct

Reply:
Personally I think this behavior is very unexpected, and I think decimals should not be supported at all.
Moreover, this is not the number of 1 bits in the actual internal representation, but in the conversion of the decimal to a long.

Reply:
I did some more testing with MySQL.
* bitcount(18446744073709551615) = 64
* bitcount(18446744073709551616 or greater) = 63
* bitcount(-9223372036854775808 or less) = 1

I have updating the implementation of `bitcount(BigDecimal)` to match this behaviour. This is only enabled in the MySQL library.

It should match MySQL behaviour at the moment, but since this is not really documented
https://dev.mysql.com/doc/refman/8.4/en/bit-functions.html#function_bit-count
perhaps this is not worth keeping and could change at any time.

In general is it better to match MySQL behaviour (for a MySQL specific function) or is it better to take the simpler approach?